aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1557,7 +1557,7 @@ inline void baddbmm_cpu_kernel(const Tensor& result, const Tensor& self, const T
                 r += s2[k] * m1[k][j];
               }
             } else {
-              r *= beta;
+              r = beta == scalar_t(0) ? scalar_t(0) : beta * r;
```

Comment:
And not that it matters that much given that this implementation will not be particularly efficient, but you may want to do this up front calling `result.zero_()` in this case and putting an if here, so that this part is vectorized. Similar to how it's done for this path:
https://github.com/pytorch/pytorch/blob/5b2ab0dd4f9b75657ba0e83641cd91864b4f23d1/aten/src/ATen/native/LinearAlgebra.cpp#L1639-L1641

Reply:
I don't think it will get a better performance calling ```result.zero_()```, for this path, the ```m*n*k``` is a smaller number, and if we use ```result.zero_()```, there has two steps(no fusion, and also have dispatch overhead).

Reply:
I do some performance test for ```m=1, n=1, k=300```:

1. threads = 20

- this PR
```
batch_size = 1, avg time is 0.006 (ms)
batch_size = 20, avg time is 0.011 (ms) 
batch_size = 40, avg time is 0.011 (ms)
batch_size = 80, avg time is 0.012 (ms)
batch_size = 160, avg time is 0.015 (ms)
```
- your proposal 
```
batch_size = 1, avg time is 0.013 (ms)
batch_size = 20, avg time is 0.019 (ms)
batch_size = 40, avg time is 0.018 (ms)
batch_size = 80, avg time is 0.023 (ms)
batch_size = 160, avg time is 0.029 (ms)
```
2. threads =1
- this PR
```
batch_size = 1, avg time is 0.013 (ms)
batch_size = 20, avg time is 0.024 (ms)
batch_size = 40, avg time is 0.036 (ms)
batch_size = 80, avg time is 0.069 (ms)
batch_size = 160, avg time is 0.137 (ms)
```
- your proposal  
```
batch_size = 1, avg time is 0.010 (ms)
batch_size = 20, avg time is 0.024 (ms)
batch_size = 40, avg time is 0.038 (ms)
batch_size = 80, avg time is 0.072 (ms)
batch_size = 160, avg time is 0.142 (ms)
```

test code:

```
import torch
num_iter = 600

def fn(a, b, c):
    return torch.baddbmm(a, b, c, beta=0.00)

for batch_size  in [1, 20, 40, 80, 160]:
    m = 1
    n = 1
    p = 300
    a = torch.randn((batch_size, n, p))
    b = torch.randn((batch_size, n, m))
    c = torch.randn((batch_size, m, p))

    fwd = 0
    with torch.no_grad():
        for i in range(300):
            y = fn(a, b, c)

    with torch.no_grad():
        t1  = time.time()
        for i in range(num_iter):
            y = fn(a, b, c)
        t2 = time.time()
        fwd = fwd + (t2 - t1)

    avg_time = fwd / num_iter * 1000
    print("batch_size = %d, avg time is %0.3f (ms) fps:%f"%(batch_size, avg_time, batch_size  * num_iter / fwd))
```